core/src/main/java/org/radargun/stages/test/Stressor.java

```diff
@@ -309,4 +314,54 @@ public Request getRequest() {
          return request;
       }
    }
+
+   static class UniformRateLimiter {
+      long start = Long.MIN_VALUE;
+      final int opsPerSec;
+      long opIndex = 0;
+
+      UniformRateLimiter(int opsPerSec) {
+         this.opsPerSec = opsPerSec;
+      }
+
+      /* @return intended start time in ns for the operation */
+      long acquire() {
+         long currOpIndex = opIndex++;
+         return start + currOpIndex * 1_000_000_000 / opsPerSec;
+      }
+
+      void start() {
+         start = TimeService.nanoTime();
+      }
+   }
+
+   // TODO maybe a better name ?
+   static class StreamOfOperation {
```

Comment:
Why are this and `UniformRateLimiter` separate classes? You could easily inline the functionality.